common/views/components/IIIFViewer/parts/MainViewer.js

```diff
@@ -53,22 +66,41 @@ const ItemRenderer = memo(({ style, index, data }) => {
   const urlTemplateMain = mainImageService['@id']
     ? iiifImageTemplate(mainImageService['@id'])
     : null;
-  const thumbnailService = currentCanvas.thumbnail.service;
-  const urlTemplateThumbnail = iiifImageTemplate(thumbnailService['@id']);
-  const smallestWidthImageDimensions = thumbnailService.sizes
-    .sort((a, b) => a.width - b.width)
-    .find(dimensions => dimensions.width > 100);
+  const thumbnailService = currentCanvas?.thumbnail?.service;
+  const urlTemplateThumbnail =
+    thumbnailService && iiifImageTemplate(thumbnailService['@id']);
+  const smallestWidthImageDimensions =
+    thumbnailService &&
+    thumbnailService.sizes
+      .sort((a, b) => a.width - b.width)
+      .find(dimensions => dimensions.width > 100);
   const infoUrl =
     mainImageService['@id'] && convertIiifUriToInfoUri(mainImageService['@id']);
   const matching = rotatedImages.find(canvas => canvas.canvasIndex === index);
   const rotation = matching ? matching.rotation : 0;
   const imageType = scrollVelocity >= 1 ? 'thumbnail' : 'main';
+  const imageAuthService = getImageAuthService(currentCanvas);
+  const isRestricted =
+    imageAuthService &&
+    imageAuthService.profile === 'http://iiif.io/api/auth/0/login/restricted';
   return (
     <div style={style}>
       {scrollVelocity === 3 || isProgrammaticScroll ? (
         <div style={{ display: 'flex', justifyContent: 'center' }}>
           <LL lighten={true} />
         </div>
+      ) : isRestricted ? (
+        <MessageContainer>
```

Comment:
should this `MessageContainer` be wrapped in `{imageAuthService && (…)}`, then the `h2` and `p` could also be conditionally rendered based on whether `label` and `description` are there?